### DISCUSSION TITLE
fish: Expected a string, but found a redirection

### DISCUSSION THREAD

**Discussion by Randommist** [fish-shell/fish-shell#9886] [Q&A] [answered]
When I used bash I used to create files like this `>filename`, but in fish I get `fish: Expected a string, but found a redirection`. What does this mean?

**Comment by faho** [comment, marked as answer]
Fish does not support running redirections without a command. This is an awkward special case in bash's grammar (actually the POSIX shell grammar) without any real need, and so it's not something fish does.

Use `touch filename` or `: >filename`. The latter of which is what bash will do implicitly.

One upside of not having this special case is that it allows you to find errors. E.g. if you accidentally do `foo & >file` instead of `foo >file &` that will error out rather than running  `foo` with the wrong stdout.
